core/internal/viewstate/view-state.go

```diff
@@ -101,21 +72,14 @@ type State interface {
 	FinishViewChange(newView uint64) (ok bool, release func())
 }
 
-// In the normal mode, if lastNewView == lastActiveView. Otherwise, in
-// the view-change mode, and lastNewView > lastActiveView.
 type viewState struct {
 	sync.Mutex
 
-	// Last active view number
-	lastActiveView uint64
+	currentView   uint64
+	expectedView  uint64
+	requestedView uint64
```

Comment:
Even we increment a view 1 million times per second, 585 thousand years are needed to consume `uint64` value. So we don't need to consider exhausting these values now (and also in the future).

Reply:
Good observation :+1: